src/KeyVault/KeyVault/Commands/Secret/GetAzureKeyVaultSecret.cs

```diff
@@ -184,33 +200,50 @@ public class GetAzureKeyVaultSecret : KeyVaultCmdletBase
             ParameterSetName = InputObjectByVaultNameParameterSet,
             HelpMessage = "Specifies whether to show the previously deleted secrets in the output.")]
         [Parameter(Mandatory = false,
-            ParameterSetName = ResourceIdByVaultNameParameterSet,
+            ParameterSetName = ParentResourceIdByVaultNameParameterSet,
+            HelpMessage = "Specifies whether to show the previously deleted secrets in the output.")]
+        [Parameter(Mandatory = false,
+            ParameterSetName = BySecretUriParameterSet,
             HelpMessage = "Specifies whether to show the previously deleted secrets in the output.")]
         public SwitchParameter InRemovedState { get; set; }
 
         [Parameter(Mandatory = false, ParameterSetName = BySecretNameParameterSet, HelpMessage = "When set, the cmdlet will convert secret in secure string to the decrypted plaintext string as output.")]
         [Parameter(Mandatory = false, ParameterSetName = ByVaultNameParameterSet)]
+        [Parameter(Mandatory = false, ParameterSetName = BySecretUriParameterSet)]
         [Parameter(Mandatory = false, ParameterSetName = InputObjectBySecretNameParameterSet)]
         [Parameter(Mandatory = false, ParameterSetName = InputObjectByVaultNameParameterSet)]
-        [Parameter(Mandatory = false, ParameterSetName = ResourceIdBySecretNameParameterSet)]
-        [Parameter(Mandatory = false, ParameterSetName = ResourceIdByVaultNameParameterSet)]
+        [Parameter(Mandatory = false, ParameterSetName = ParentResourceIdBySecretNameParameterSet)]
+        [Parameter(Mandatory = false, ParameterSetName = ParentResourceIdByVaultNameParameterSet)]
         public SwitchParameter AsPlainText { get; set; }
         #endregion
 
         public override void ExecuteCmdlet()
         {
             PSKeyVaultSecret secret;
 
+            // Check input object
             if (InputObject != null)
             {
                 VaultName = InputObject.VaultName.ToString();
             }
-            else if (!string.IsNullOrEmpty(ResourceId))
+            else if (!string.IsNullOrEmpty(ParentResourceId))
+            {
+                var parsedParentResourceId = new ResourceIdentifier(ParentResourceId);
+                VaultName = parsedParentResourceId.ResourceName;
+            }
+
+            // Handle SecretId (uri) parameter
+            if (ParameterSetName == BySecretUriParameterSet)
             {
-                var parsedResourceId = new ResourceIdentifier(ResourceId);
-                VaultName = parsedResourceId.ResourceName;
+                var secretUri = new Uri(Id);
```

Comment:
what will happen if we input Id as a random string without uri format? we need to pre-validate its format and throw informational message to customer.

Reply:
This is the Error Message in the case that the -Id is malformed.

`Get-AzKeyVaultSecret: Invalid URI: The format of the URI could not be determined.`